Receive files outside the classpath in quarkus-liquibase

juanjgarciavilla opened the pull request:
This PR is the continuation of an improvement made in quarkus-liquibase-mongodb extension which was replicated in quarkus-liquibase extension, more information about it can be found in this [issue](https://github.com/quarkusio/quarkus/issues/39683).

juanjgarciavilla committed:
Uploading dynamic resource accessor implementation according to changelog location

uploading an adjustment

adding ClassLoaderResourceAccessor to the compositeResourceAccessor

formatting code

geoand commented on extensions/liquibase-mongodb/runtime/src/main/java/io/quarkus/liquibase/mongodb/LiquibaseMongodbFactory.java:
```diff
@@ -51,6 +51,8 @@ private ResourceAccessor resolveResourceAccessor() throws FileNotFoundException
         }
 
         CompositeResourceAccessor compositeResourceAccessor = new CompositeResourceAccessor();
+        compositeResourceAccessor
+                .addResourceAccessor(new ClassLoaderResourceAccessor(Thread.currentThread().getContextClassLoader()));
```
Why this change?

juanjogv replied:
It was something I realized along the way that it can be useful to have a changelog within the classpath that references files on and off the classpath as shown below.

![image](https://github.com/quarkusio/quarkus/assets/51928477/e079081a-b0e1-45cd-b735-defec653c638)

geoand replied:
Thanks for the information.

geoand replied:
In this case I think this change should be a separate commit as it's pretty much unrelated to the rest of the changes in PR

juanjogv replied:
Understood, but then I put it in a different commit or in a different pr?

juanjgarciavilla committed:
Adding the classpath to the CompositeResourceAccessor searchpath

Added the classpath to the CompositeResourceAccessor of LiquibaseMongodbFactory, this way even if you specify a custom searchpath you will still be able to reference some file inside the resources folder.

geoand reviewed (approved):
Thanks!

geoand commented on extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/runtime/LiquibaseDataSourceBuildTimeConfig.java:
```diff
@@ -28,4 +32,9 @@ public static final LiquibaseDataSourceBuildTimeConfig defaultConfig() {
+    /**
+     * The search path for DirectoryResourceAccessor
+     */
+    @ConfigItem(defaultValue = "/")
```
Yeah I saw that but I have to admit it's confusing. Anyway, let's leave it like this for now, but I assume that using `Optional<List<String>>` for `searchPath` would be better.

juanjogv replied:
I did it this way because at the time I could not find how to declare the default list as an empty list in the build time properties, because if I put a default value of "" in the tag, that is an empty string, I got an error related to null.

geoand replied:
```java
    /**
     * The search path for DirectoryResourceAccessor
     */
    @ConfigItem
    public Optional<List<String>> searchPath;
```

should work, no?

geoand replied:
Let's merge this and you can try and update both configs with my suggestion. If all goes well, you can open a new PR
